File: solutions/461.hamming-distance/hamming-distance.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function(x, y) {
    let hamming = x ^ y;
    let res = 0;
    while(hamming!==0){
        res++;
        hamming &= (hamming - 1);
    }
    return res;
};

File: solutions/461.hamming-distance/hamming-distance.java
class Solution {
    public int hammingDistance(int x, int y) {
        int hamming = x ^ y;
        int res = 0;
        for(int i = 0; i < 31; i++){
            if((hamming & 1) == 1){
                res++;
            }
            hamming >>= 1;
        }
        return res;
    }
}